### PR TITLE
JAMES-3085 Drop direct references to ReactorRabbitMQChannelPool

### PR DESCRIPTION
**Pull request by blackheaven**
ReactorRabbitMQChannelPool was introduced in order to limit our
RabbitMQ's channels usage. It was set to 5.

We have 3 kinds of usage of it:

 * Ever running Receivers (4)
 * Temporary Senders (2)
 * Management operations (3)

*Ever running Receivers* hold a channel forever and are allocated at
startup.
*Management operations* need a channel but do not need them exclusively.
The real benefit of ReactorRabbitMQChannelPool is for the *Temporary
senders* which request a channel each time.

Before this commit, only one channel was available for *Management
operations* and *Temporary senders*, leading to starvations and timeouts
under heavy loads.

This commit make the *Ever running Receivers* relying directly on a
channel. The other usages relying (indirectly) on
ReactorRabbitMQChannelPool (with a smaller channel pool).

Now 7 channels are open (4 for the *Ever running Receivers* and
3 for the ReactorRabbitMQChannelPool) instead of 5.

### CODE REVIEW

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/RabbitMQHealthCheck.java [hunk 1]
```diff
@@ -48,7 +46,7 @@ public ComponentName componentName() {
     @Override
     public Result check() {
         try {
-            if (connectionPool.tryConnection() && rabbitChannelPoolImpl.tryChannel()) {
```
Comment (chibenwa) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/RabbitMQHealthCheck.java]: We don't try acquiring a channel anymore?

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReceiverProvider.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/**
+ * *************************************************************
```
Comment (chibenwa) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReceiverProvider.java]: Check all your licenses...

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java [hunk 3]
```diff
@@ -103,7 +103,7 @@ public void destroyObject(PooledObject<Channel> pooledObject) throws Exception {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ReactorRabbitMQChannelPool.class);
     private static final long MAXIMUM_BORROW_TIMEOUT_IN_MS = Duration.ofSeconds(5).toMillis();
-    private static final int MAX_CHANNELS_NUMBER = 5;
+    private static final int MAX_CHANNELS_NUMBER = 3;
```
Comment (chibenwa) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java]: Why do we decrease this parameter?

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/RabbitMQHealthCheck.java (hunk 1 above)
Comment (blackheaven) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/RabbitMQHealthCheck.java]: I'm not sure it still make sense

**Comment by Arsnael** [comment]
```
[4c4c4f8a247672d6ee1495bbacf4694be70f22e9] [ERROR] src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java:[29,8] (imports) UnusedImports: Unused import - javax.inject.Inject.
```

**Comment by Arsnael** [comment]
```
[feb292c69eb6b36826afc9ff43de7219d218cdb4] [ERROR] Errors: 
[feb292c69eb6b36826afc9ff43de7219d218cdb4] [ERROR]   DistributedTaskManagerTest.additionalInformationShouldBeAvailableOnAnyTaskManagerDuringExecution » ConditionTimeout
[feb292c69eb6b36826afc9ff43de7219d218cdb4] [ERROR]   DistributedTaskManagerTest.givenOneEventStoreTwoEventTaskManagersShareTheSameEvents:200 » ConditionTimeout
[feb292c69eb6b36826afc9ff43de7219d218cdb4] [ERROR]   DistributedTaskManagerTest.givenTwoTaskManagerATaskRunningOnOneShouldBeCancellableFromTheOtherOne:260 » ConditionTimeout
[feb292c69eb6b36826afc9ff43de7219d218cdb4] [ERROR]   DistributedTaskManagerTest.givenTwoTaskManagerATaskSubmittedOnOneCouldBeRunOnTheOther:242 » ConditionTimeout
[feb292c69eb6b36826afc9ff43de7219d218cdb4] [ERROR]   DistributedTaskManagerTest.givenTwoTaskManagerAndATaskRanPerTaskManagerListingThemOnEachShouldShowBothTasks:318 » ConditionTimeout
[feb292c69eb6b36826afc9ff43de7219d218cdb4] [ERROR]   DistributedTaskManagerTest.givenTwoTaskManagerIfTheFirstOneIsDownTheSecondOneShouldBeAbleToRunTheRemainingTasks:342 » ConditionTimeout
[feb292c69eb6b36826afc9ff43de7219d218cdb4] [ERROR]   DistributedTaskManagerTest.givenTwoTaskManagersATaskRunningOnOneShouldBeWaitableFromTheOtherOne:289 » ConditionTimeout
[feb292c69eb6b36826afc9ff43de7219d218cdb4] [ERROR]   DistributedTaskManagerTest.givenTwoTaskManagersAndTwoTaskOnlyOneTaskShouldRunAtTheSameTime:227 » ConditionTimeout
[feb292c69eb6b36826afc9ff43de7219d218cdb4] [ERROR]   RabbitMQWorkQueuePersistenceTest.submittedMessageShouldSurviveRabbitMQRestart:86->startNewWorkqueue:97 » AlreadyClosed
```

**Comment by Arsnael** [comment]
Read it

**Comment by blackheaven** [comment]
Rebased to add the DNR

**Comment by blackheaven** [comment]
Rebased for RRCP size for tests

**Comment by blackheaven** [comment]
Rebase to set the right channel pool usage.

File: server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java [hunk 4]
```diff
@@ -104,20 +104,20 @@ private void startWorkqueue() {
 
     @VisibleForTesting
     void declareQueue() {
-        channelPool.getSender()
+        sender
```
Comment (remk) [inline server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueue.java]: Not rebased on latest version of #3169

**Comment by blackheaven** [comment]
Rebased to update the DNR commit

**Comment by blackheaven** [comment]
ObjectStorageBlobStoreAWSTest random failure

test this please

**Comment by blackheaven** [comment]
test this please

**Comment by nvduc91** [comment]
Read it

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**